src/ZODB/Connection.py

```diff
@@ -757,7 +758,11 @@ def afterCompletion(self, transaction):
         #    finalizing previous ones without calling begin.  We pass
         #    False to avoid possiblyt expensive sync calls to not
         #    penalize well-behaved applications that call begin.
-        self.newTransaction(transaction, False)
+        if hasattr(self._storage, 'afterCompletion'):
+            self._storage.afterCompletion()
```

Comment:
I don't understand what you're saying.    Are you referring to the lines above? or below? (I'm guessing below.(

We're only deferring in the explicit case.

I would think that the cost of polling remote servers outweighs the benefits of maybe invalidating things.

I agree that doing a cache gc here makes sense.

I also wonder about the wisdom of doing cache gc at the *beginning* of a transaction.  Perhaps the cache gc should be moved from new to here.